pontoon/sync/core/translations_to_repo.py

```diff
@@ -0,0 +1,221 @@
+import logging
+
+from collections import defaultdict
+from collections.abc import Container
+from datetime import datetime
+from os import remove
+from os.path import commonpath, isfile, join, normpath
+
+from moz.l10n.paths import L10nConfigPaths, L10nDiscoverPaths
+
+from django.conf import settings
+from django.db.models import Q
+from django.db.models.manager import BaseManager
+
+from pontoon.base.models import Locale, Project, Translation, User
+from pontoon.base.models.changed_entity_locale import ChangedEntityLocale
+from pontoon.sync.core.checkout import Checkouts
+from pontoon.sync.formats import parse
+from pontoon.sync.formats.po import POResource
+from pontoon.sync.repositories import CommitToRepositoryException, get_repo
+
+
+log = logging.getLogger(__name__)
+
+
+def sync_translations_to_repo(
+    project: Project,
+    commit: bool,
+    locale_map: dict[str, Locale],
+    checkouts: Checkouts,
+    paths: L10nConfigPaths | L10nDiscoverPaths,
+    db_changes: BaseManager[ChangedEntityLocale],
+    changed_source_paths: set[str],
+    removed_source_paths: set[str],
+    now: datetime,
+) -> None:
+    readonly_locales = project.locales.filter(project_locale__readonly=True)
+    removed = delete_removed_resources(
+        project, paths, locale_map, readonly_locales, removed_source_paths
+    )
+    updated, updated_locales, translators = update_changed_resources(
+        project,
+        paths,
+        locale_map,
+        readonly_locales,
+        db_changes,
+        changed_source_paths,
+        now,
+    )
+    if not removed and not updated:
+        return
+
+    if not commit:
+        log.info(f"[{project.slug}] Skipping commit & push")
+        return
+
+    if removed:
+        lc_str = "all localizations"
+    else:
+        if len(updated_locales) > 4:
+            lc_str = f"{len(updated_locales)} localizations"
+        else:
+            lc_str = ", ".join(f"{loc.name} ({loc.code})" for loc in updated_locales)
+    commit_msg = f"Pontoon/{project.name}: Update {lc_str}"
```

Comment:
I'm not a fan of this commit message.

```
Pontoon/Firefox (local): Update Italian (it)
```

I find the existing one more readable:

```
Pontoon: Update Italian (it) localization of Firefox (local)
```

Reply:
Have you considered what the message looks like if it contains changes for multiple locales? With the old format, it quickly gets much more variable and clunkier.

Reply:
Pontoon only commits one locale at a time now. Is that changing with the new code?

Reply:
I'm guessing yes, given `lc_str = "1 locale" if lc_count == 1 else f"{lc_count} locales"`.

I feel like we have discussed this at some point, but it's been long enough that I don't know if that happened, or where 🤔

Reply:
Resolving this.